wlauto/workloads/uxperf_applaunch/__init__.py

```diff
@@ -0,0 +1,160 @@
+    captured by the parametre *workload_params*. This information can be obtained by
+    inspecting the workload details of the specific workload.
+
+    The workload allows to run multiple iterations of an application
+    launch in two modes:
+
+    1. Launch from background
+    2. Launch from long-idle
+
+    These modes are captured as a parameter applaunch_type.
+
+    *Launch from background*
+        Launches an application after the application is sent to background by
+        pressing Home button.
+
+    *Launch from long_idle*
+        Launches an application after killing an application process and
+        clearing all the caches.
+
+    **Test Description:**
+
+    1. During the initialization and setup, the application being launched is
+    launched for the first time. The jar file of the workload of the application
+    is moved to device at the location specified a parameter *workload_jar_path*
+    which further implements the methods needed to measure the
+    application launch time.
+
+    2. Run phase calls the UiAutomator of the uxperfapplaunch which runs in two subphases.
+        A. Applaunch Setup Run:
+            During this phase, welcome screens and dialogues during the first launch
+            of the instrumented application are cleared.
+        B. Applaunch Metric Run:
+            During this phase, the application is launched multiple times determined by
+            the iteration number specified by the parametre *applaunch_iterations*.
+            Each of these iterations are instrumented to capture the launch time taken
+            and the values are recorded as UXPERF marker values in logfile.
+    '''
+    supported_platforms = ['android']
+
+    parameters = [
+        Parameter('workload_name', kind=str,
+                  description='Name of the uxperf workload to launch',
+                  mandatory=True),
+        Parameter('workload_params', kind=dict, default={},
+                  description="""
+                  parameters of the uxperf workload whose application launch
+                  time is measured
+                  """),
+        Parameter('applaunch_type', kind=str, default='launch_from_background',
+                  allowed_values=['launch_from_background', 'launch_from_long-idle'],
+                  description="""
+                  Choose launch_from_long-idle for measuring launch time
+                  from long-idle. These two types are described in the class
+                  description.
+                  """),
+        Parameter('applaunch_iterations', kind=int, default=1,
+                  description="""
+                  Number of iterations of the application launch
+                  """),
+    ]
+
+    def __init__(self, device, **kwargs):
+        super(UxperfApplaunch, self).__init__(device, **kwargs)
+        #set working directory.
+        self.device.working_directory = "/data/local/tmp"
```

Comment:
No. The working directory is set by the user; the workloads must not mess with it.